userspace/falco/app/actions/load_config.cpp

```diff
@@ -18,15 +18,67 @@ limitations under the License.
 #include "actions.h"
 #include "falco_utils.h"
 
+/* DEPRECATED: we will remove it in Falco 0.38. */
+#define FALCO_BPF_ENV_VARIABLE "FALCO_BPF_PROBE"
+
 using namespace falco::app;
 using namespace falco::app::actions;
 
-// applies legacy/in-deprecation options to the current config
-static void apply_deprecated_options(
-		const falco::app::options& opts,
-		const std::shared_ptr<falco_configuration>& cfg)
+// applies legacy/in-deprecation options to the current state
+static falco::app::run_result apply_deprecated_options(falco::app::state& s)
 {
-	// Keep for future use cases.
+	// Please note: is not possible to mix command line options and configs to obtain a configuration
+	// we need to use only one method. For example, is not possible to set the gvisor-config through
+	// the command line and the gvisor-root through the config file. For this reason, if we detect
+	// at least one change in the default config we don't allow to use the command line options.
+	if(s.config->m_changes_in_engine_config)
+	{
+		return run_result::ok(); 
+	}
+
+	// Replace the kmod default values in case the engine was open with the kmod.
+	// We don't have a command line option to open the kmod so we have to always enforce the
+	// default values.
+	s.config->m_kmod.m_drop_failed_exit = s.config->m_syscall_drop_failed_exit;
+	s.config->m_kmod.m_buf_size_preset = s.config->m_syscall_buf_size_preset;
+
+	// If overridden from CLI options (soon to be removed),
+	// use the requested driver.
+	if (getenv(FALCO_BPF_ENV_VARIABLE))
```

Comment:
What will happen when [this](https://github.com/falcosecurity/falco/pull/2918) gets merged, and perhaps some of the users will use the same env var in the config :point_down: ? 

```yaml
engine:
  kind: kmod
  kmod:
    buf_size_preset: 4
    drop_failed_exit: false
  ebpf:
    # path to the elf file to load.
    probe: ${FALCO_BPF_PROBE}
```

:exploding_head:

Reply:
> Nothing, i mean the deprecated env variable takes precedence.

Will it print a warning even if it shouldn't? :thinking:

Reply:
Yes it will print the newly introduced warning about the fact that FALCO_BPF_PROBE env variable will be ignored, and that is correct(since we won't directly use the env variable but the new config key that resolves to the env variable value)

Reply:
Oh i mis-read (i thought `kind: bpf` was in use). In this case, as i stated earlier, the deprecated env variable takes precedence (since the `bpf` kind configuration is not even parsed, because the `kind` is kmod).
No warning in this case (except for the `FALCO_BPF_PROBE` env var deprecation )